code/modules/photography/camera/camera.dm

```diff
@@ -0,0 +1,220 @@
+
+#define CAMERA_PICTURE_SIZE_HARD_LIMIT 21
+
+/obj/item/camera
+	name = "camera"
+	icon = 'icons/obj/items_and_weapons.dmi'
+	desc = "A polaroid camera."
+	icon_state = "camera"
+	item_state = "electropack"
+	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
+	w_class = WEIGHT_CLASS_SMALL
+	flags_1 = CONDUCT_1
+	slot_flags = ITEM_SLOT_BELT
+	materials = list(MAT_METAL = 50, MAT_GLASS = 150)
+	var/state_on = "camera"
+	var/state_off = "camera_off"
+	var/pictures_max = 10
+	var/pictures_left = 10
+	var/on = TRUE
+	var/cooldown = 64
+	var/blending = FALSE		//lets not take pictures while the previous is still processing!
+	var/see_ghosts = CAMERA_NO_GHOSTS //for the spoop of it
+	var/obj/item/disk/holodisk/disk
+	var/sound/custom_sound
+	var/silent = FALSE
+	var/picture_size_x = 1
+	var/picture_size_y = 1
+	var/picture_size_x_min = 0
+	var/picture_size_y_min = 0
+	var/picture_size_x_max = 3
+	var/picture_size_y_max = 3
+
+/obj/item/camera/CheckParts(list/parts_list)
+	..()
+	var/obj/item/camera/C = locate(/obj/item/camera) in contents
+	if(C)
+		pictures_max = C.pictures_max
+		pictures_left = C.pictures_left
+		visible_message("[C] has been imbued with godlike power!")

```

Comment:
??? Even if this is about some blessing craft, this is assuming everything done on cameras does this.